aten/src/ATen/native/TensorProperties.cpp

```diff
@@ -53,11 +53,37 @@ Tensor & detach_(Tensor & self) {
 }
 
 Tensor contiguous(const Tensor & self) {
-  if (self.is_contiguous()) {
-    return self;
-  }
-  return self.clone();
+  return contiguous(self, MemoryFormat::Contiguous);
```

Comment:
Minor nit: is this resulting in a double-dispatch, or does it call straight to the native implementation? I think it's the native implementation.

Reply:
it calls native